src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -200,36 +176,34 @@ private static SonarRule CreateRule(JObject r, JToken actives)
         ///
         public async Task<IDictionary<string, string>> GetProperties(string projectKey, string projectBranch)
         {
-            if (string.IsNullOrWhiteSpace(projectKey))
-            {
-                throw new ArgumentNullException(nameof(projectKey));
-            }
+            Contract.ThrowIfNullOrWhitespace(projectKey, nameof(projectKey));
 
             var projectId = GetProjectIdentifier(projectKey, projectBranch);
 
-            if (await IsSonarCloud() || (await GetServerVersion()).CompareTo(new Version(6, 3)) >= 0)
-            {
-                return await GetComponentProperties(projectId);
-            }
-            else
-            {
-                return await GetComponentPropertiesLegacy(projectId);
-            }
+            return await IsSonarCloud() || (await GetServerVersion()).CompareTo(new Version(6, 3)) >= 0
+                       ? await GetComponentProperties(projectId)
+                       : await GetComponentPropertiesLegacy(projectId);
         }
 
         public async Task<Version> GetServerVersion()
         {
             if (serverVersion == null)
             {
-                await DownloadServerVersion();
+                var uri = GetUri("api/server/version");
+                serverVersion = await ExecuteWithLogs(async () =>
+                {
+                    var contents = await downloader.Download(uri);
+                    var separator = contents.IndexOf('-');
+                    return separator >= 0 ? new Version(contents.Substring(0, separator)) : new Version(contents);
```

Comment:
Can be simplified a lot
```suggestion
                    return new Version(contents.Split('-').First());
```